krabs/krabs/filtering/event_filter.hpp

```diff
@@ -73,9 +82,9 @@ namespace krabs {
         template <typename U>
         void add_on_event_callback(const U &callback);
 
-        unsigned short provider_filter_event_id() const
+        std::vector<unsigned short> provider_filter_event_ids() const
```

Comment:
Return a `const std::vector<unsigned short>&` or else you're making a copy of the vector.